src/main/webapp/app/exam/exam-scores/exam-scores.component.ts

```diff
@@ -703,4 +767,19 @@ export class ExamScoresComponent implements OnInit, OnDestroy {
     logErrorOnSentry(errorMessage: string) {
         Sentry.captureException(new Error(errorMessage));
     }
+
+    private static declareExerciseType(exerciseInfo: ExerciseInfo) {
+        switch (exerciseInfo.exerciseType) {
+            case 'TextExercise':
+                return ExerciseType.TEXT;
+            case 'ModelingExercise':
+                return ExerciseType.MODELING;
+            case 'ProgrammingExercise':
+                return ExerciseType.PROGRAMMING;
+            case 'FileUploadExercise':
+                return ExerciseType.FILE_UPLOAD;
+            case 'QuizExercise':
+                return ExerciseType.QUIZ;
+        }
```

Comment:
[nit] I think we should create a build/make/construct function for `ExerciseType` enum.
Something like: `ExerciseType.fromExerciseInfo(exerciseInfo: ExerciseInfo) { ... }`

It seems like in typescript one would do it like this: https://stackoverflow.com/questions/28150739/add-functions-to-an-enum
What do you think?

Reply:
How about simply creating a `createExerciseTypeByExerciseInfo` method in the `exercise.model.ts`? This way we still have the enum